### PR TITLE
 Add certs handler for kube entity

### PR DESCRIPTION
**Pull request by saheienko**
This change adds an ability to retrieve a key/cert bundle from a kubernetes master.

### CODE REVIEW

File: pkg/kube/certs.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+)
+
+// Bundler represents a key/certificate pair.
+type Bundle struct {
+	Cert []byte
+	Key  []byte
+}
+
+// Certs handles keys and certificates for kube components.
+type Certs struct {
```
Comment (yegorch) [inline pkg/kube/certs.go]: Is it possible to make it an interface to mock it out in tests?

File: pkg/kube/certs.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	return &Bundle{
+		Cert: crtBytes,
+		Key:  crtKey,
+	}, nil
+}
+
+func (c *Certs) getFile(ctx context.Context, path string) ([]byte, error) {
+	stdout, stderr := &bytes.Buffer{}, &bytes.Buffer{}
```
Comment (yegorch) [inline pkg/kube/certs.go]: Using bash to read a file from go code is ... questionable practice. Please rewrite it using standard golang functions e.g. iotuil.ReadFile, etc.

File: pkg/kube/certs_test.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+	"github.com/supergiant/supergiant/pkg/runner"
+	"github.com/supergiant/supergiant/pkg/runner/command"
+	"github.com/supergiant/supergiant/pkg/runner/shell"
+)
+
+var (
+	fakeErrFileNotFound = errors.New("key file not found")
+)
+
+type fakeRunner struct {
```
Comment (yegorch) [inline pkg/kube/certs_test.go]: Why not just outright mock it using testify mock package?

File: pkg/kube/certs.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		return nil, err
+	}
+
+	return &Bundle{
+		Cert: crtBytes,
+		Key:  crtKey,
+	}, nil
+}
+
+func (c *Certs) getFile(ctx context.Context, path string) ([]byte, error) {
```
Comment (yegorch) [inline pkg/kube/certs.go]: Can you please rename this method? Since it is intended to fetch certs from the k8s node the name of it is misleading as it's looks like it tries to read a local file using runner. 
Make it something like readCertificateFiles

Comment (stgleb) [inline pkg/kube/certs.go]: why you call these writers as std? this looks confusing since std is only for standard file descriptors that are passed to an application.

File: pkg/kube/certs.go [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+	stdout, stderr := &bytes.Buffer{}, &bytes.Buffer{}
+	cmd := command.NewCommand(ctx, "cat", []string{path}, stdout, stderr)
+
+	err := c.r.Run(cmd)
+	if err != nil {
+		return nil, errors.Wrapf(err, "run %q", cmd.FullCommand())
+	}
```
Comment (stgleb) [inline pkg/kube/certs.go]: Are you sure that some output on stderr means that command was failed? Some application uses stderr as a default for log output(logrus).

Comment (yegorch) [reply pkg/kube/certs.go]: @stgleb :+1:  the only reliable way to determine if remote bash script has failed is to check it's exit code, if it's non zero - it's failed.

File: pkg/runner/runner.go [hunk 6]
```diff
@@ -4,5 +4,5 @@ import "github.com/supergiant/supergiant/pkg/runner/command"
 
 // Runner is interface for running command in different environment
 type Runner interface {
-	Run(command command.Command)
```
Comment (stgleb) [inline pkg/runner/runner.go]: I would like to do that with separate pull request https://github.com/supergiant/supergiant/pull/528

File: pkg/kube/certs_test.go (hunk 3 above)
Comment (saheienko) [reply pkg/kube/certs_test.go]: Its pretty hard to make it with a testify mock. You can update this tests later.

File: pkg/kube/service.go [hunk 7]
```diff
@@ -151,6 +153,36 @@ func (s *Service) GetKubeResources(ctx context.Context, kname, resource, ns, nam
 	return raw, nil
 }
 
+// GetCerts returns a keys bundle for provided component name.
+func (s *Service) GetCerts(ctx context.Context, kname, cname string) (*Bundle, error) {
+	kube, err := s.Get(ctx, kname)
+	if err != nil {
+		return nil, errors.Wrap(err, "storage: get")
```
[the remark this answers is not in the document]
Comment (saheienko) [reply pkg/kube/service.go]: It's already returns this error:
```
	raw, err := s.storage.Get(ctx, s.prefix, name)
	if err != nil {
		return nil, errors.Wrap(err, "storage: get")
	}
	if raw == nil {
		return nil, sgerrors.ErrNotFound
	}
```
Better to update a wrapper message...

File: pkg/kube/certs.go (hunk 4 above)
Comment (saheienko) [reply pkg/kube/certs.go]: Nope, it's not intended for certificates only. This is a generic func and it can retrieve any file on the provided path.

File: pkg/kube/certs.go (hunk 5 above)
Comment (saheienko) [reply pkg/kube/certs.go]: You are right. I've just actually used current implementations of the ssh client.
@yegorch If you have any ideas about retrieving an exit code here, you are welcome.

File: pkg/runner/runner.go (hunk 6 above)
Comment (saheienko) [reply pkg/runner/runner.go]: Yeap, I'll sync with it.

File: pkg/kube/certs.go (hunk 4 above)
Comment (stgleb) [reply pkg/kube/certs.go]: yes, but it can be named just out, because standard output is namely standard FD that is created per process.
